src/main/java/uk/gov/hmcts/ccd/domain/service/globalsearch/GlobalSearchParser.java

```diff
@@ -0,0 +1,92 @@
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.casedetails.SecurityClassification;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.search.global.SearchCriteria;
+import uk.gov.hmcts.ccd.domain.service.common.AccessControlService;
+import uk.gov.hmcts.ccd.domain.service.common.CaseTypeService;
+import uk.gov.hmcts.ccd.domain.service.common.SecurityClassificationService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+@Component
+@Slf4j
+public class GlobalSearchParser {
+
+    private static final String FIELD_SEPARATOR = ".";
+    private final UserRepository userRepository;
+    private final CaseTypeService caseTypeService;
+    private final SecurityClassificationService securityClassificationService;
+
+    @Autowired
+    public GlobalSearchParser(@Qualifier(CachedUserRepository.QUALIFIER)
+                                  UserRepository userRepository,
+                              CaseTypeService caseTypeService,
+                              SecurityClassificationService securityClassificationService) {
+        this.userRepository = userRepository;
+        this.caseTypeService = caseTypeService;
+        this.securityClassificationService = securityClassificationService;
+    }
+
+    public List<CaseDetails> filterCases(List<CaseDetails> results, SearchCriteria request) {
+        List<String> fields = findFieldsToFilter(request);
+        results.removeIf(caseDetails -> !authorised(fields, caseDetails));
+
+        return results;
+    }
+
+    private boolean authorised(List<String> fields, CaseDetails caseDetails) {
+        CaseTypeDefinition caseTypeDefinition = caseTypeService.getCaseType(caseDetails.getCaseTypeId());
+        boolean condition = true;
+        Optional<SecurityClassification> userClassificationOpt =
+            securityClassificationService.getUserClassification(caseDetails.getJurisdiction());
+        for (String field : fields) {
+            Optional<CaseFieldDefinition> caseFieldDefinition =
+                (field.contains(FIELD_SEPARATOR)) ? caseTypeDefinition.getComplexSubfieldDefinitionByPath(field)
+                    : caseTypeDefinition.getCaseField(field);
+            if (caseFieldDefinition.isPresent() && (!AccessControlService
+                .hasAccessControlList(userRepository.getUserRoles(), CAN_READ,
+                    caseFieldDefinition.get().getAccessControlLists()))){
+                condition = false;
+                break;
```

Comment:
```suggestion
                return false;
```
Do we need to `break`? If there's no processing in this method afterwards then we can just return it here.